### ISSUE TITLE
XPS 95570 On Debian 10 Buster, problem to switch from Nvidia GPU to Intel GPU

### ISSUE THREAD

**Issue by Phil995511** [delta-one/dell-xps-15-9570#1]
Hello, I have the same machine as you, a Dell XPS 9570 / i7-8750H / Nvidia GTX 1050 Ti / FHD non-touch screan / Intel 9260 lan card on Debian 10 Buster Cinnamon with integrated firmwares non-free + Kernel 4.19 64 Bits + Sharkoon 3-Port USB 3.0 Aluminium Hub with RJ 45 Ethernet Adapter. I was tired of Ubuntu and its incessant updates, I switched to Debian 10 as on my workstation. I followed your tutorial for Optimus / Nvidia but I can't get it to work ;( I found others but again it doesn't work !? You seem to be using a kernel more recent than me, is that the reason for my non-success ? You use the Nvidia 418 or 430 backport drivers ?? You use again the Killer 1535 lan card ?? Killer 1535 is very unstable for me. Intel 9260 offers better wifi / bluetooth transmission, but the XPS bios is designed to use the Killer 1535, it supports the Intel 9260 a little less (turn on / off the card). Regards

**Comment by delta-one** [comment]
1.) Are you using stable, testing or unstable ?

2.) Which kernel and which Nvidia-driver are you using ? The important thing is, that the Nvidia-module compiles with the kernel.

3.) What exactly does not work regarding Optimus ? What is the error message ?

4.) I had a Killer card, but switched to the Intel card, because the connection wasn't the best with the Killer card. But then I switched back to the Killer card, because the Intel card became pretty unusable (Wifi crahsed frequently and became quite slow). But now the connection with the Killer card seems much better than it was before and I plan to stick with it for the time being.
As far as I remember, you don't need to change anything in the BIOS when you switch the Wifi-chip. You just need a kernel, that has the corresponding modules enabled. The default Debian-kernel should have both drivers installed as modules. If you want to be sure, you can doublecheck it with `cat /boot/config-$(uname -r) | grep CONFIG_IWLWIFI=` and `cat /boot/config-$(uname -r) | grep CONFIG_ATH10K=`. In both cases, the option should be set to `m`.

**Comment by Phil995511** [comment]
Hello,
 
 Thank you for your message !!
 
 1) I use Debian 10 Buster stable 64 bits + Cinnamon. DKMS are installed on my system.
 
 2) I use a kernel version 4.19.0-8-amd64 (the kernel 5.40 are present on backports but it's not @ 100 % stable with display) and nvidia drivers version 418.74-1 (430 are present on backports but not tested by my)
 
 3) I have a multiple problems 
 
 3.1) I must to run your scripts as root
 
 3.2)  philippe@portable-tests:~$ glxinfo | grep "OpenGL renderer"
OpenGL renderer string: Mesa DRI Intel(R) UHD Graphics 630 (Coffeelake 3x8 GT2) 
philippe@portable-tests:~$ optirun glxinfo | grep "OpenGL renderer"
primus: fatal: failed to load any of the libraries: /usr/lib/x86_64-linux-gnu/nvidia/libGL.so.1:/usr/lib/i386-linux-gnu/nvidia/libGL.so.1:/usr/lib/nvidia/libGL.so.1
/usr/lib/x86_64-linux-gnu/nvidia/libGL.so.1: cannot open shared object file: No such file or directory
/usr/lib/i386-linux-gnu/nvidia/libGL.so.1: cannot open shared object file: No such file or directory
/usr/lib/nvidia/libGL.so.1: cannot open shared object file: No such file or directory
philippe@portable-tests:~$ glxgears
Running synchronized to the vertical refresh.  The framerate should be
approximately the same as the monitor refresh rate.
392 frames in 5.0 seconds = 78.171 FPS
301 frames in 5.0 seconds = 60.002 FPS
300 frames in 5.0 seconds = 59.999 FPS
^C
philippe@portable-tests:~$ optirun glxgears
primus: fatal: failed to load any of the libraries: /usr/lib/x86_64-linux-gnu/nvidia/libGL.so.1:/usr/lib/i386-linux-gnu/nvidia/libGL.so.1:/usr/lib/nvidia/libGL.so.1
/usr/lib/x86_64-linux-gnu/nvidia/libGL.so.1: cannot open shared object file: No such file or directory
/usr/lib/i386-linux-gnu/nvidia/libGL.so.1: cannot open shared object file: No such file or directory
/usr/lib/nvidia/libGL.so.1: cannot open shared object file: No such file or directory
4) For me, the Intel card does the job much better than the Killer card.
Best regards and thank you again for your kindness !!!
Philippe
 
  ----Message d'origine----
  
De : notifications@github.com
  
Date : 04/03/2020 - 00:53 (E)
  
À : dell-xps-15-9570-debian-linux@noreply.github.com
  
 
Objet : Re: [delta-one/dell-xps-15-9570-debian-linux] XPS 95570 On Debian 10 Buster, problem to switch from Nvidia GPU to Intel GPU (#1)
  
  
  
1.) Are you using stable, testing or unstable ?
  
2.) Which kernel and which Nvidia-driver are you using ?
  
3.) What exactly does not work regarding Optimus ? What is the error message ?
  
4.) I had a Killer card, but switched to the Intel card, because the connection wasn't the best with the Killer card. But then I switched back to the Killer card, because the Intel card became pretty unusable (Wifi crahsed frequently and became quite slow). But now the connection with the Killer card seems much better than it was before and I plan to stick with it for the time being.
 As far as I remember, you don't need to change anything in the BIOS when you switch the Wifi-chip. You just need a kernel, that has the corresponding modules enabled. The default Debian-kernel should have both drivers installed as modules. If you want to be sure, you can doublecheck it with cat /boot/config-$(uname -r) | grep CONFIG_IWLWIFI= and cat /boot/config-$(uname -r) | grep CONFIG_ATH10K=. In both cases, the option should be set to m.

**Comment by delta-one** [comment]
3.1) Yes, you must run the scripts as root. Unfortunately there is no way around it.

3.2) It seems, that you are missing a library. I cannot test it on stable, since I run testing, but what is the output of `locate libGL.so.1` and `apt-file search libGL.so.1` ? You might have to install `locate` and `apt-file` first, but they are good to have anyway.

**Comment by Phil995511** [comment]
Hello,
Thanks for your mail.
I did not know the two software which you advise me to use, I will test them and index them for a futur utilisation.
The big problem is that it seems that there are "libGL.so.1" modules are installed in many places under Debian according to the research that I could do. Solving this problem is not going to be easy... I do not know if I will be able to find the problem, if I succeed, I will inform you.
A track to follow will be this :
https://superuser.com/questions/1404875/how-to-dowload-libgl-so-1
I also tried another method :
https://debian-facile.org/viewtopic.php?pid=295019#p295019
But arrived at the end of this one I have another error when I launch the command "optirun glxgears" :
optirun glxinfo
[   71.858702] [ERROR]Cannot access secondary GPU - error: Could not enable discrete graphics card
[   71.858746] [ERROR]Aborting because fallback start is disabled.
It's really difficult to get intel / nvidia cards to work under Debian !!! They should integrate nvidia-prime from Ubuntu that will make life easier for people !!!
The XPS 9570 is really great for doing laboratory tests. The possibility of installing OS on the internal ssd, on the SD card reader or on external USB 3 / USB C / thunderbolt ssd, and to be able to boot on it by configuring the bios of this machine is just magnificent...
I use the TimeShift software to take images disk and restore them (by a USB key with ISO of Mint LMDE4).
I don't know if this site works or not, I had to test 4 without success ... including one from Debian; (
https://github.com/delta-one/dell-xps-15-9570-debian-linux
https://www.dell.com/support/article/en-au/sln298475/pr%C3%A9sentation-de-la-vid%C3%A9o-hybride-sur-un-ordinateur-dell-%C3%A9quip%C3%A9-du-syst%C3%A8me-d-exploitation-ubuntu-en-anglais?lang=fr
https://wiki.debian.org/NvidiaGraphicsDrivers/Optimus
https://unix.stackexchange.com/questions/320384/how-to-install-nvidia-driver-on-debian-and-switch-between-drivers
justpaste.it/1q9vo
If on your side you come across a miracle solution to my problems let me know.
Thanks for trying to help me, I'm going to try one or two more methods and I think I'm going to give up, it's taking me too long and it doesn't work...
https://wiki.debian-fr.xyz/NVidia_Optimus_Bumblebee
https://debian-facile.org/doc:materiel:cartes-graphique:nvidia:optimus
If only we could just and simply blacklist the card that we don't want to use when booting the machine ...
Best regards.
Philippe
----Message d'origine----
De : notifications@github.com
Date : 04/03/2020 - 20:06 (E)
À : dell-xps-15-9570-debian-linux@noreply.github.com
Cc : xxx@yyy.com, author@noreply.github.com
Objet : Re: [delta-one/dell-xps-15-9570-debian-linux] XPS 95570 On Debian 10 Buster, problem to switch from Nvidia GPU to Intel GPU (#1)
3.1) Yes, you must run the scripts as root. Unfortunately there is no way around it.
3.2) It seems, that you are missing a library. I cannot test it on stable, since I run testing, but what is the output of locate libGL.so.1 and apt-file search libGL.so.1 ? You might have to install locate and apt-file first, but they are good to have anyway.

**Comment by delta-one** [comment]
@Phil995511 Have you made any progress ?

**Comment by Phil995511** [comment]
Good evening Patrick,

Thank you for your message, unfortunately I did not manage to make the
Nvidia driver of my laptop work, despite all the methods that I could find
on the internet that I tested, as well as the attempts at reflection that I
have done on this topic.

Best regards.

Philippe

Le jeu. 19 mars 2020 à 00:19, Patrick Franz <notifications@github.com> a
écrit :

**Comment by delta-one** [comment]
You could still try, what I wrote earlier:

1.) Install `locate` and `apt-file`.
2.) Run `locate libGL.so.1` and `apt-file search libGL.so.1`.

**Comment by Phil995511** [comment]
I have to take care of my daughter these last days whose school is closed,
I do not have time to try to make this graphics card work.

And then I tried everything that was humanly possible to make it work
without getting there.

Some sites speak of an impossible bug to solve, it seems to be linked to
the 1050 Ti card. Other graphics cards appear to be working properly.

I had used optimus on my old laptop years ago, without much difficulty, but
this time it seems really difficult to solve. Even the solutions proposed
on the Nvidia forum do not work ;(

Regards.

Le ven. 20 mars 2020 à 16:24, Patrick Franz <notifications@github.com> a
écrit :

**Comment by delta-one** [comment]
Alright, good luck.

I'll probably re-install my computer in a couple of weeks and will update the NVIDIA section then. Feel free to either re-open this issue or create a new one, should you find the time to look into it again.

**Comment by Phil995511** [comment]
Hello Patrick, thank you for your message. I will not hesitate to come and
look at your github page in futur to have information about the clean
installation of Debian Stable that you plan to carry out. Make attention to
yourself in these troubled times when the coronavirus is present among us.
Best regards. Philippe

Le dim. 22 mars 2020 à 13:59, Patrick Franz <notifications@github.com> a
écrit :

**Comment by delta-one** [comment]
Got some good news for you. I did a clean install of stable with KDE Plasma as the desktop environment (though that should probably not really matter) and got the NVIDIA card working by exactly following the steps described. It also worked for stable with backports and testing. I adjusted the description slightly.

**Comment by Phil995511** [comment]
Hello Patrick,

Thank you for your message.

I have making a new clean installation of Debian 10.3 for more testing your
methodology.

1 - Install the following packages : apt install bumblebee-nvidia
primus-nvidia nvidia-smi => primus-nvidia not exist on Debian 10, the
correct name of the concerned packet is `primus`

2 - Similarly, if you are using powertop, you might have to
disable/blacklist the card there as well.
=> /etc/default/powertop> RUNTIME_PM_BLACKLIST="01:00.0" It's correct ??

3 - When you make your 2 scritps `enableGPU.sh` and `disableGPU.sh` it
seems to me that you have to give the rights in to run these files as a
program, right ??

4 - optirun glxinfo | grep " OpenGL renderer " > [149.505008] [ERROR]
Cannot access secondary GPU - error: Could not load GPU driver
> [149.405042] [ERROR] Aborting because fallback start is disable

5- When I use `enableGPU.sh` =>

optirun glxinfo | grep "OpenGL renderer"
primus: fatal: failed to load any of the libraries:
/usr/lib/x86_64-linux-gnu/nvidia/libGL.so.1:/usr/lib/i386-linux-gnu/nvidia/libGL.so.1:/usr/lib/nvidia/libGL.so.1
/usr/lib/x86_64-linux-gnu/nvidia/libGL.so.1: cannot open shared object
file: No such file or directory
/usr/lib/i386-linux-gnu/nvidia/libGL.so.1: cannot open shared object file:
No such file or directory
/usr/lib/nvidia/libGL.so.1: cannot open shared object file: No such file or
directory

Conclusion your tuto is not fonctional for my despite the changes you
tell me you have made ;(

Regards.

Philippe


Le sam. 11 avr. 2020 à 04:24, Patrick Franz <notifications@github.com> a
écrit :

**Comment by delta-one** [comment]
I still don't understand, why it goes wrong for you. I tested it 3 times on 10.3.

* Sorry about the `primus-nvidia`, it only exists in Testing at the moment, but installiing `primus` should pull all the dependencies as well.
* I don't quite understand what you are doing with the powertop.
* Yes, you have to make the scripts executable (`chmod +x enableGPU.sh`) and run them as root.
* Is `libgl-dev` installed ?
* What is the output when you run `enableGPU.sh` ?

**Comment by Phil995511** [comment]
For more efficiency tests, I installed KDE Plasma, but the result is similar, it doesn't work.

If I install the 5.4 kernel, the result is similar, it doesn't work.

I do not understand what is not working given that we are supposed to have exactly the same machine and the same installation of Debian !?

Regards.

**Comment by delta-one** [comment]
* Is `libgl-dev` installed ?
* What is the output when you run `enableGPU.sh` ?

**Comment by Phil995511** [comment]
I have desactived powertop, but I don't understand your recommendation for activate it, it's not clear, so I made a guess...

`libgl-dev` come with the package `libgl1-mesa-dev` now I install it... because it's not installed from base.

Now with `libgl1-mesa-dev` installed I have the following messages

disableGPU.sh >

~$ glxinfo | grep "OpenGL renderer"
OpenGL renderer string: Mesa DRI Intel(R) UHD Graphics 630 (Coffeelake 3x8 GT2) 
~$ optirun glxinfo | grep "OpenGL renderer"
[  180.087358] [ERROR]Cannot access secondary GPU - error: Could not load GPU driver

[  180.087391] [ERROR]Aborting because fallback start is disabled.

enableGPU.sh >

~$ glxinfo | grep "OpenGL renderer"
OpenGL renderer string: Mesa DRI Intel(R) UHD Graphics 630 (Coffeelake 3x8 GT2) 
~$ optirun glxinfo | grep "OpenGL renderer"
[  217.488328] [ERROR]Cannot access secondary GPU - error: Could not load GPU driver

[  217.488341] [ERROR]Aborting because fallback start is disabled.

**Comment by delta-one** [comment]
What is printed in your terminal, when you run `enableGPU.sh` ?

**Comment by Phil995511** [comment]
Yes after installation of `libgl1-mesa-dev`

**Comment by delta-one** [comment]
What is the output, when you run the command `enableGPU.sh` ?

**Comment by Phil995511** [comment]
First I must to execute `disableGPU.sh` for obtain a resultat.

After this, when I execute `EnableGPU.sh` a message say :

"Allowing to load Nvidia modules...
Charging power control..."

After a moment a message display :

" NVIDIA-SMI has failed because it couldn't communicate with teh NVIDIA driver. Make sure that the lasted NVIDIA driver is installed and running" 

If i use Synaptic and I make a search for "Ndidia-driver" the version 418.74-1 has been correctly installed !?

You want me to make a `apt install -t buster-backports nvidia-driver` ??

Regards

**Comment by delta-one** [comment]
You don't need the backports. 

Honestly, I don't know why things go wrong for you. Maybe you made a mistake in one of the steps. You could ask a friend to doublecheck the steps you took.

As for the latest error, you can check `/var/log/syslog` for error messages.

**Comment by Phil995511** [comment]
I started to follow your tutorial from the chapter "Video card".

Should we perhaps apply "Kernel parameters" ??

I check the parameters of install :

I do not understand why, suddenly primus-nvidia is available for my laptop in `version 0~20150328-10~bpo10+1` !? It must come from backports... I install It, but It not change the result ;(

All is ok for the another parameters ;( This is really a crazy story.

 The last time I tried, I also set everything correctly, I also checked everything and I had the same problems, it didn't work !?

All the other tutorials I have found on the Internet also fail.

My syslog file is attached to this message if you can find a significant error

Regards.

[syslog.txt](https://github.com/delta-one/dell-xps-15-9570-debian-linux/files/4494610/syslog.txt)

PS After tests all run perfectly on Mint/Ubuntu the problem is only present on Debian.


...

**Comment by delta-one** [comment]
What are you running ? Stable or backports ? Install the kernel headers for the backports-kernel. If that doesn't work, ask a friend to doublecheck, what you did. You need a fresh set of eyes.

**Comment by Phil995511** [comment]
As I described to you I made a stable installation, it did not work I then
tested the kernel 5.4 therefore backports for kernel only, if I use the
drivers 418 I am on stable.

I don't need a friend to doublecheck my job, I'm not a beginner, I have 37
years of IT under my belt. Thank you again for your advice but your
procedure simply does not work.

Regards

Le sam. 18 avr. 2020 à 00:09, Patrick Franz <notifications@github.com> a
écrit :
